akd_local_auditor/src/storage/s3/mod.rs

```diff
@@ -53,11 +64,36 @@ pub struct S3ClapSettings {
         value_parser = is_bucket_name_valid
     )]
     bucket: String,
+
+    /// The AWS region to operate in
+    #[clap(long)]
+    region: String,
+
+    /// [OPTIONAL] An custom URI for the AWS endpoint
+    #[clap(long, value_parser = is_uri_valid)]
+    endpoint: Option<String>,
```

Comment:
Any reason not to make this an `Option<URI>` to make a slightly stronger type than just `String`?

Reply:
Again a Clap thing. Uri isn't implementing Debug, parser, etc which are all necessary traits to parse the clap args properly

Reply:
Makes sense! Thanks for clarifying :)